packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -34,101 +36,102 @@ public class AmplifyStorageOperations {
+                        }
+                        listResultDictionary["items"] = storageItemList
+                        flutterResult(listResultDictionary);
+                    case.failure(let storageError):
+                        prepareError(flutterResult: flutterResult, storageError: storageError)
+                    }
+             })
+        } catch {
+            prepareGenericError(flutterResult: flutterResult, error: error)
         }
-        
     }
     
     public static func downloadFile(flutterResult: @escaping FlutterResult, request: Dictionary<String, AnyObject>){
-        if(FlutterDownloadFileRequest.isValid(request: request)) {
+        do {
+            try FlutterDownloadFileRequest.validate(request: request)
             let req = FlutterDownloadFileRequest(request: request)
             _ = Amplify.Storage.downloadFile(key: req.key,
-                                             local: req.file,
-                                             options: req.options,
-                                             resultListener: { event in
-                                                switch event {
-                                                case .success:
-                                                    var result = [String: String]()
-                                                    // Amplify Android sends this back
-                                                    result["path"] = req.file.absoluteURL.path
-                                                    flutterResult(result)
-                                                case .failure(let storageError):
-                                                    prepareError(flutterResult: flutterResult, msg: FlutterStorageErrorMessage.DOWNLOAD_FILE_FAILED.rawValue, localizedError: storageError.errorDescription, recoverySuggestion: storageError.recoverySuggestion)
-                                                }
+                 local: req.file,
+                 options: req.options,
+                 resultListener: { event in
+                    switch event {
+                    case .success:
+                        var result = [String: String]()
+                        // Amplify Android sends this back
+                        result["path"] = req.file.absoluteURL.path
+                        flutterResult(result)
+                    case .failure(let storageError):
+                        prepareError(flutterResult: flutterResult, storageError: storageError)
+                    }
             })
-        } else {
-            prepareError(flutterResult: flutterResult, msg: FlutterStorageErrorMessage.DOWNLOAD_FILE_MALFORMED.rawValue, localizedError: "The request received was malformed", recoverySuggestion: "Please ensure the request matches the method signature")
+        } catch {
+            prepareGenericError(flutterResult: flutterResult, error: error)
```

Comment:
Do we need the `prepareError` and `prepareGenericError` as separate methods in iOS? We have only `prepareError` in Android.

Reply:
Fixed this.